utils/build.ps1

```diff
@@ -1096,6 +1096,40 @@ function Get-Dependencies {
     Expand-Archive -Path $source -DestinationPath $destination -Force
   }
 
+  function Expand-TarBall {
+    param
+    (
+        [string]$TarBallFileName,
+        [string]$BinaryCache,
+        [string]$ExtractPath,
+        [bool]$CreateExtractPath = $true
+    )
+    $source = Join-Path -Path $BinaryCache -ChildPath $TarBallFileName
+    $destination = Join-Path -Path $BinaryCache -ChildPath $ExtractPath
+
+    # Check if the extracted directory already exists and is up to date.
+    if (Test-Path $destination) {
+        $zipLastWriteTime = (Get-Item $source).LastWriteTime
+        $extractedLastWriteTime = (Get-Item $destination).LastWriteTime
+        # Compare the last write times
+        if ($zipLastWriteTime -le $extractedLastWriteTime) {
+            Write-Output "'$TarBallFileName' is already extracted and up to date."
+            return
+        }
+    }
+
+    $destination = if ($CreateExtractPath) { $destination } else { $BinaryCache }
+
+    Write-Output "Extracting '$TarBallFileName' ..."
+    New-Item -ItemType Directory -ErrorAction Ignore -Path $destination | Out-Null
+    try {
+        tar -xvf $source -C $destination | Out-Null
```

Comment:
Will this actually ever throw?

Reply:
This should not. I thought that if `tar` returned `1` the function would throw but that's not the case. Fixed, thanks!